python/ray/data/_internal/plan.py

```diff
@@ -541,6 +543,8 @@ def execute_to_iterator(
         except StopIteration:
             pass
         self._snapshot_stats = executor.get_stats()
+        # Store executor stats getter to update _snapshot_stats during execution.
+        self._get_executor_stats = executor.get_stats
         return block_iter, self._snapshot_stats, executor
```

Comment:
`test_can_pickle` is failing because we store this here. But `Dataset` can store `_current_executor`, what's the difference?